config.js

```diff
@@ -504,19 +551,114 @@ exports.updateFromArgv = function() {
         }
     }
 
-    if (argv.timecontrol) {
-        for (let i of argv.timecontrol.split(',')) {
-            exports.allowed_timecontrols[i] = true;
+    if (argv.boardsizeranked) {
+        for (let boardsizeranked of argv.boardsizeranked.split(',')) {
+            if (boardsizeranked == "all") {
+                exports.allow_all_sizes_ranked = true;
+            } else if (boardsizeranked == "custom") {
+                exports.allow_custom_sizes_ranked = true;
+                for (let boardsizewidthranked of argv.boardsizewidthranked.split(',')) {
+                    exports.allowed_custom_boardsizewidth_ranked[boardsizewidthranked] = true;
+                }
+                for (let boardsizeheightranked of argv.boardsizeheightranked.split(',')) {
+                    exports.allowed_custom_boardsizeheight_ranked[boardsizeheightranked] = true;
+                }
+            } else {
+                exports.allowed_sizes_ranked[boardsizeranked] = true;
+            }
         }
     }
 
+    if (argv.boardsizeunranked) {
+        for (let boardsizeunranked of argv.boardsizeunranked.split(',')) {
+            if (boardsizeunranked == "all") {
+                exports.allow_all_sizes_unranked = true;
+            } else if (boardsizeunranked == "custom") {
+                exports.allow_custom_sizes_unranked = true;
+                for (let boardsizewidthunranked of argv.boardsizewidthunranked.split(',')) {
+                    exports.allowed_custom_boardsizewidth_unranked[boardsizewidthunranked] = true;
+                }
+                for (let boardsizeheightunranked of argv.boardsizeheightunranked.split(',')) {
+                    exports.allowed_custom_boardsizeheight_unranked[boardsizeheightunranked] = true;
+                }
+            } else {
+                exports.allowed_sizes_unranked[boardsizeunranked] = true;
+            }
+        }
+    }
+
+    if (argv.komi) {
+        for (let komi of argv.komi.split(',')) {
+            if (komi == "all") {
```

Comment:
Please use `===` to compare strings in ogs. It's so much safer since it will typecheck too. `==` in js means "about equal" and can result in all kind of (obscure) problems. It's just good practice to always use `===`

Reply:
thanks for mentionning :+1: @Dorus 

even the beginner that i am read that this is not recommended, but seeing as the code was like that, i just stupidly copy pasted it without trying to mess what is already working, and it works

if we want to start that, there are a big lot of `==` in the code, so i would rather have you or someone else code enhance all the ogs code on a separate commit after this is merged

after all, fixing one thing while keeping the other one would still have the problem present :1234: